docs/file-formats/colors.md

```diff
@@ -2,26 +2,123 @@
-| `comment` | `string` | No       | An optional description of the color, explaining contextual information, such as how it should be used. This may be displayed in the Lona Studio UI. This will not appear in the generated code.                                  |
+| Attribute  | Type      | Required                           | Description                                                                                                                                                                                                                       |
+| ---------- | --------- | ---------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `id`       | `string`  | Yes                                | The unique `id` of the color, which will be used internally by Lona Studio and in the generated code. This should be _code-friendly_: it must not contain spaces or unusual characters, since it will be used as a variable name. |
+| `name`     | `string`  | Yes                                | The human-readable name of the color. This will be displayed in the Lona Studio UI. This will not appear in the generated code.                                                                                                   |
+| `value`    | `object`  | Required if there is no `variants` | An object representing the color. See [Color Value](#color-value) for more information about the properties of the object.                                                                                                        |
+| `variants` | `Color[]` | Required if there is no `value`    | An array representing the variants of the color.                                                                                                                                                                                  |
+| `comment`  | `string`  | No                                 | An optional description of the color, explaining contextual information, such as how it should be used. This may be displayed in the Lona Studio UI. This will not appear in the generated code.                                  |
+
+> The reason why it's an array instead of an object with the id as keys is that we want to order colors, and it's really not recommended to rely on the order of the keys of a JSON object.
+
+### Color Value
+
+#### Value
+
+The value of a color is represented by a JSON object. There are multiple ways to represent it, hence there are multiple set of properties available to represent the color.
+
+- `hex`: `{ "hex": "#00FF00" }`
+- `rgb`: `{ "r": 0, "g": 1, "b": 0 }`
```

Comment:
wondering if we should use `r`, `g`, `b` or `red`, `green`, `blue`. Might not matter too much for `rgb` but could be clearer for `hsl`

Reply:
The issue with CSS is that it's pretty limited: you only have 255*255*255 values, there is no color profile support, etc. I wouldn't be surprised if there is an extension to CSS sooner than later around wide gamut screen support and that kind of things.

The compiler can easily translate a more complete color value description down to CSS if needed.

Reply:
That makes sense, I didn't know about the new CSS color spec. In that case, howabout we support the proposed syntax? https://www.w3.org/TR/css-color-4/#changes-from-3

We don't have to support it all at once, but can start with the things that seem most important/relevant for us